src/newshop.cpp

```diff
@@ -276,9 +278,16 @@ bool uninstall_ware_from_target_character(struct obj_data *obj, struct char_data
     obj_from_bioware(obj);
     GET_INDEX(victim) -= GET_CYBERWARE_ESSENCE_COST(obj);
```

Comment:
This is where the character's bioware index is updated. Putting your essence cost change after this means that that change has no effect. Move your essence cost ifcheck before this instead.

Also, can you change this to `GET_BIOWARE_ESSENCE_COST` instead of cyberware? They mean the same thing, but this typo makes it harder to read.

Reply:
Swapped to CYBER, shoulda realized it sooner they're the same thing